src/module/actor/creature/index.ts

```diff
@@ -709,24 +709,34 @@ export abstract class CreaturePF2e extends ActorPF2e {
         const rollOptions = this.getRollOptions(selectors);
 
         if (movementType === "land") {
-            const label = game.i18n.localize("PF2E.SpeedTypesLand");
-            const base = Number(systemData.attributes.speed.value ?? 0);
-            const statLabel = game.i18n.format("PF2E.SpeedLabel", { type: label });
+            const landSpeed = systemData.attributes.speed;
+            landSpeed.value = Number(landSpeed.value) || 0;
+
+            const fromSynthetics = (this.synthetics.movementTypes[movementType] ?? []).map((d) => d() ?? []).flat();
```

Comment:
Seems like this call is duplicated, you can probably lift it out of the if/else. And I guess the "get best logic" as well can probably be lifted out.

Reply:
It can be handled in a separate refactor PR tbh. Having this fix in before a release is more important.